Algorithm.Python/HourReverseSplitRegressionAlgorithm.py

```diff
@@ -0,0 +1,42 @@
+﻿# QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+# Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from clr import AddReference
+AddReference("System")
+AddReference("QuantConnect.Algorithm")
+AddReference("QuantConnect.Indicators")
+AddReference("QuantConnect.Common")
+
+from System import *
+from QuantConnect import *
+from QuantConnect.Algorithm import *
+from QuantConnect.Indicators import *
+from datetime import datetime, timedelta
+
+### <summary>
+### Regression test for consistency of hour data over a reverse split event in US equities.
+### </summary>
+### <meta name="tag" content="using data" />
+### <meta name="tag" content="regression test" />
+class HourReverseSplitRegressionAlgorithm(QCAlgorithm):
+
+    def Initialize(self):
+        self.SetStartDate(2013, 11, 7)
+        self. SetEndDate(2013, 11, 8)
+        self.SetCash(100000)
+        self.symbol = self.AddEquity("VXX", Resolution.Hour).Symbol
+    
+    def OnData(self, tradeBars):
+        if tradeBars.HasData:
```

Comment:
Since the tradeBars object is actuall a Slice object, we could use slice or data. Otherwise it can confuse users. Also, instead of HasData, we better use 
`if slice.Bars.Count == 0: return` 
Because we actually have data and the Buy order is being placed earlier than the regression test requires to.